File: src/main/java/kr/pe/ecmaxp/thumbsj/InterruptHandler.java
package kr.pe.ecmaxp.thumbsj;

import kr.pe.ecmaxp.thumbsj.exc.InvalidMemoryException;
import kr.pe.ecmaxp.thumbsj.signal.ControlPauseSignal;
import kr.pe.ecmaxp.thumbsj.signal.ControlStopSignal;

public interface InterruptHandler
{
    void invoke(int imm) throws InvalidMemoryException, ControlStopSignal, ControlPauseSignal;
}

File: src/main/java/kr/pe/ecmaxp/thumbsj/InterruptHandler.kt
package kr.pe.ecmaxp.thumbsj

import kr.pe.ecmaxp.thumbsj.exc.InvalidMemoryException
import kr.pe.ecmaxp.thumbsj.signal.ControlPauseSignal
import kr.pe.ecmaxp.thumbsj.signal.ControlStopSignal

interface InterruptHandler {
    @Throws(InvalidMemoryException::class, ControlStopSignal::class, ControlPauseSignal::class)
    operator fun invoke(imm: Int)
}
